src/observations/tracklet.jl

```diff
@@ -0,0 +1,182 @@
+"""
+@auto_hash_equals struct Tracklet{T <: AbstractFloat}
+    radec::Vector{RadecMPC{T}}
+    observatory::ObservatoryMPC{T}
+    night::TimeOfDay
+    date::DateTime
+    α::T
+    δ::T
+    v_α::T
+    v_δ::T
+    mag::T
+    nobs::Int
+    indices::Vector{Int}
+    # Inner constructor
+    function Tracklet{T}(radec::Vector{RadecMPC{T}}, observatory::ObservatoryMPC{T},
+                                 night::TimeOfDay, date::DateTime, α::T, δ::T, v_α::T, v_δ::T,
+                                 mag::T, nobs::Int, indices::Vector{Int}) where {T <: AbstractFloat}
+
+        return new{T}(radec, observatory, night, date, α, δ, v_α, v_δ, mag, nobs, indices)
+    end
+end
+
+# Functions to get specific fields of a Tracklet object
+date(x::Tracklet{T}) where {T <: AbstractFloat} = x.date
+ra(x::Tracklet{T}) where {T <: AbstractFloat} = x.α
+dec(x::Tracklet{T}) where {T <: AbstractFloat} = x.δ
+observatory(x::Tracklet{T}) where {T <: AbstractFloat} = x.observatory
+vra(x::Tracklet{T}) where {T <: AbstractFloat} = x.v_α
+vdec(x::Tracklet{T}) where {T <: AbstractFloat} = x.v_δ
+mag(x::Tracklet{T}) where {T <: AbstractFloat} = x.mag
+indices(x::Tracklet{T}) where {T <: AbstractFloat} = x.indices
+
+# Print method for Tracklet{T}
+# Examples:
+# 3 observation tracklet around 2023-11-18T19:59:55.392 at WFST, Lenghu
+# 4 observation tracklet around 2023-11-22T08:07:46.336 at Mt. Lemmon Survey
+function show(io::IO, x::Tracklet{T}) where {T <: AbstractFloat}
+    print(io, x.nobs, " observation tracklet around ", x.date, " at ", x.observatory.name)
+end
+
+# Order in Tracklet is given by date
+isless(a::Tracklet{T}, b::Tracklet{T}) where {T <: AbstractFloat} = a.date < b.date
+
+# Evaluate a polynomial with coefficients p in every element of x
+polymodel(x, p) = map(y -> evalpoly(y, p), x)
+
+# Normalized mean square residual for polynomial fit
+polyerror(x) = sum(x .^ 2) / length(x)
+
+@doc raw"""
+    polyfit(x::Vector{T}, y::Vector{T}; tol::T = 1e-4) where {T <: AbstractFloat}
+
+Fit a polynomial to points `(x, y)`. The order of the polynomial is increased
+until `polyerror` is less than `tol`.
+"""
+function polyfit(x::Vector{T}, y::Vector{T}; tol::T = 1e-4) where {T <: AbstractFloat}
+    # Avoid odd and high orders (to have a defined concavity and avoid overfit)
+    for order in [1, 2, 4, 6]
+        # Initial guess for coefficients
+        coeffs = ones(T, order+1)
+        # Polynomial fit
+        fit = curve_fit(polymodel, x, y, coeffs)
+        # Convergence condition
+        if polyerror(fit.resid) < tol || order == 6
+            return fit.param
+        end
+    end
+end
+
+@doc raw"""
+    diffcoeffs(x::Vector{T}) where {T <: AbstractFloat}
+
+Return the coefficients of the derivative of a polynomial with coefficients `x`.
+"""
+function diffcoeffs(x::Vector{T}) where {T <: AbstractFloat}
+    y = Vector{T}(undef, length(x)-1)
+    for i in eachindex(y)
+        y[i] = i * x[i+1]
+    end
+    return y
+end
```

Comment:
Have you considered using Polynomials.jl for this functionality? If Polynomials.jl has similar functionalities (fitting a polynomial model to an univariate time series) it's probably worth it giving it a try

Reply:
I think it's better to have `LsqFit.jl` as a dependency, in case we ever need to fit a function other than a polynomial. For instance, a future implementation of ADS (see https://github.com/PerezHz/TaylorIntegration.jl/pull/178) might require fitting exponentials.